packages/common/src/services/remote-config/remote-config.ts

```diff
@@ -82,37 +82,50 @@ export const remoteConfig = <
 
     client.onReady().then(() => {
       state.didInitialize = true
-
-      // Call initializationCallbacks
-      state.initializationCallbacks.forEach((cb) => cb())
-      state.initializationCallbacks = []
+      emitter.emit('init')
     })
   }
 
   // API
 
   /**
    * Register a callback for client ready.
-   * @param func
    */
-  function onClientReady(func: () => void) {
+  function onClientReady(cb: () => void) {
     if (state.didInitialize) {
-      func()
+      cb()
     } else {
-      state.initializationCallbacks = [...state.initializationCallbacks, func]
+      const handler = () => {
+        cb()
+        emitter.removeListener('init', handler)
+      }
+
+      emitter.addListener('init', handler)
     }
   }
 
-  // Use event emission to track setUser events
-  const emitter = new EventEmitter()
+  /**
+   * Register a callback for user ready.
+   */
+  function onUserReady(cb: () => void) {
```

Comment:
nit: since I don't think we're consuming this function outside of remote-config, maybe move it out of the `api` section and make it private?